habitat-lab/habitat/core/vector_env.py

```diff
@@ -642,3 +650,51 @@ def _spawn_workers(
             for q, read_wrapper in zip(parent_write_queues, read_fns)
         ]
         return read_fns, write_fns
+
+
+class BatchRenderedVectorEnv(VectorEnv):
```

Comment:
Not 100% convinced that class inheritance is the right solution. We will tend to get a combinatorial explosion. For example, what if the user wants to use `ThreadedVectorEnv` with batch rendering? Then you'll need a `BatchRenderedThreadedVectorEnv`.

You could consider composition instead. E.g. VectorEnv/ThreadedVectorEnv has an optional BatchRenderHelper member that gets optionally initialized based on a config value.

Reply:
Great point. This combinatorial explosion is also felt in test. I'll explore a better way to wrap this functionality.